art/estimators/classification/keras.py

```diff
@@ -442,7 +442,22 @@ def fit_generator(self, generator, nb_epochs=20, **kwargs):
         else:
             super(KerasClassifier, self).fit_generator(generator, nb_epochs=nb_epochs, **kwargs)
 
-    def get_activations(self, x, layer, batch_size):
+    @property
+    def layer_names(self):
+        """
+        Return the hidden layers in the model, if applicable.
+
+        :return: The hidden layers in the model, input and output layers excluded.
+        :rtype: `list`
+
+        .. warning:: `layer_names` tries to infer the internal structure of the model.
+                     This feature comes with no guarantees on the correctness of the result.
+                     The intended order of the layers tries to match their order in the model, but this is not
+                     guaranteed either.
+        """
+        return self._layer_names
+
```

Comment:
Please remove these line, they should not be required as they are defined in `NeuralNetworkMixin`.